src/test/scala/beam/sim/SecondaryActivitiesSpec.scala

```diff
@@ -0,0 +1,171 @@
+import beam.utils.csv.GenericCsvReader
+import com.google.inject
+import com.typesafe.config.ConfigFactory
+import org.matsim.api.core.v01.events.{ActivityEndEvent, Event}
+import org.matsim.core.api.experimental.events.EventsManager
+import org.matsim.core.events.handler.BasicEventHandler
+import org.matsim.core.scenario.{MutableScenario, ScenarioUtils}
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+class SecondaryActivitiesSpec
+    extends WordSpecLike
+    with Matchers
+    with BeamHelper
+    with GenericEventsSpec
+    with BeforeAndAfterAll {
+
+  def config: com.typesafe.config.Config =
+    ConfigFactory
+      .parseString(s"""
+                    |beam.agentsim.lastIteration = 0
+                    |beam.exchange.scenario.urbansim.activitySimEnabled = true
+                    |beam.agentsim.agents.tripBehaviors.mulitnomialLogit.generate_secondary_activities = true
+                    |beam.agentsim.agents.tripBehaviors.mulitnomialLogit.intercept_file_path = "test/test-resources/beam/agentsim/activities/activity-intercepts.csv"
+                    |beam.agentsim.agents.tripBehaviors.mulitnomialLogit.activity_file_path = "test/test-resources/beam/agentsim/activities/activity-params.csv"
+                     """.stripMargin)
+      .withFallback(testConfig("test/input/beamville/beam.conf"))
+      .resolve()
+
+  private var injector: inject.Injector = _
+  private var interceptMode: Set[String] = _
+
+  override def beforeAll(): Unit = {
+    val beamConfig = BeamConfig(config)
+    val beamScenario = loadScenario(beamConfig)
+    val configBuilder = new MatSimBeamConfigBuilder(config)
+    val matsimConfig = configBuilder.buildMatSimConf()
+    matsimConfig.planCalcScore().setMemorizingExperiencedPlans(true)
+    FileUtils.setConfigOutputFile(beamConfig, matsimConfig)
+
+    val scenario = ScenarioUtils.loadScenario(matsimConfig).asInstanceOf[MutableScenario]
+    scenario.setNetwork(beamScenario.network)
+
+    injector = org.matsim.core.controler.Injector.createInjector(
+      matsimConfig,
+      module(config, beamConfig, scenario, beamScenario)
+    )
+
+    beamServices = injector.getInstance(classOf[BeamServices])
+
+    val popAdjustment = DefaultPopulationAdjustment(beamServices)
+    popAdjustment.update(scenario)
+
+    eventManager = injector.getInstance(classOf[EventsManager])
+    networkHelper = injector.getInstance(classOf[NetworkHelper])
+
+    val interceptFilePath = beamConfig.beam.agentsim.agents.tripBehaviors.mulitnomialLogit.intercept_file_path
+    val interceptPath = Source.fromFile(interceptFilePath)
+    interceptMode = interceptPath.getLines().next().split(",").drop(1).toSet
+    interceptPath.close()
+  }
+
+  "Beam Run" should {
+
+    "check increased activities " in {
+      val eventsType = new ListBuffer[String]()
```

Comment:
`eventsType` is written from one thread and it will be read from another which can lead to undefined behavior (if it uses `LoggingEventsManager` which is most probably the case: 
https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/scala/beam/sim/LoggingEventsManager.scala#L107-L134). Either lock or thread-safe collection is required

Reply:
replace ListBuffer to thread-safe collection